specs/ExtendedProcessFailed.md

```diff
@@ -0,0 +1,671 @@
+
+* `ExitCode` is always `1` when `ProcessFailedKind` is `BROWSER_PROCESS_EXITED`, and `STILL_ACTIVE` (`259`) when `ProcessFailedKind` is `RENDER_PROCESS_UNRESPONSIVE`.
+
+
+# API Notes
+See [API Details](#api-details) section below for API reference.
+
+* Triple-slash (`///`) comments will appear in the public API documentation.
+* Double-slash (`//`) comments are notes for this review only and will not show in public documentation.
+
+
+# API Details
+## COM
+```cpp
+library WebView2
+{
+// ...
+
+/// Specifies the process failure type used in the
+/// `ICoreWebView2ProcessFailedEventHandler` interface. The values in this enum
+/// make reference to the process kinds in the Chromium architecture. For more
+/// information about what these processes are and what they do, see
+/// [Browser Architecture - Inside look at modern web browser](https://developers.google.com/web/updates/2018/09/inside-browser-part1)
+[v1_enum]
+typedef enum COREWEBVIEW2_PROCESS_FAILED_KIND {
+  // Existing stable values.
+
+  // Note: docs for this enum value remain unchanged.
+  /// Indicates that the browser process ended unexpectedly.  The WebView
+  /// automatically moves to the Closed state.  The app has to recreate a new
+  /// WebView to recover from this failure.
+  COREWEBVIEW2_PROCESS_FAILED_KIND_BROWSER_PROCESS_EXITED,
+
+  /// Indicates that the main frame's render process ended unexpectedly.  A new
+  /// render process is created automatically and navigated to an error page.
+  /// You can use the Reload method to try to reload the page that failed.
+  COREWEBVIEW2_PROCESS_FAILED_KIND_RENDER_PROCESS_EXITED,
+
+  /// Indicates that the main frame's render process is unresponsive.
+  COREWEBVIEW2_PROCESS_FAILED_KIND_RENDER_PROCESS_UNRESPONSIVE,
+
+  // New values.
+  
+  /// Indicates that a frame-only render process ended unexpectedly. The process
+  /// exit does not impact the top-level document, only a subset of the
+  /// subframes within it. The content in these frames is replaced with an error
+  /// page in the frame.
+  COREWEBVIEW2_PROCESS_FAILED_KIND_FRAME_RENDER_PROCESS_EXITED,
+
+  /// Indicates that a utility process ended unexpectedly.
+  COREWEBVIEW2_PROCESS_FAILED_KIND_UTILITY_PROCESS_EXITED,
+
+  /// Indicates that a sandbox helper process ended unexpectedly.
+  COREWEBVIEW2_PROCESS_FAILED_KIND_SANDBOX_HELPER_PROCESS_EXITED,
+
+  /// Indicates that the GPU process ended unexpectedly.
+  COREWEBVIEW2_PROCESS_FAILED_KIND_GPU_PROCESS_EXITED,
+
+  /// Indicates that a PPAPI plugin process ended unexpectedly.
+  COREWEBVIEW2_PROCESS_FAILED_KIND_PPAPI_PLUGIN_PROCESS_EXITED,
+
+  /// Indicates that a PPAPI plugin broker process ended unexpectedly.
+  COREWEBVIEW2_PROCESS_FAILED_KIND_PPAPI_BROKER_PROCESS_EXITED,
+
+  /// Indicates that a process of unspecified kind ended unexpectedly.
```

Comment:
Does anything produce this value today?

This value becomes problematic if we extend the enum to include other types of processes (`COREWEBVIEW2_PROCESS_FAILED_KIND_VIDEO_DECODE_PROCESS_EXITED`). We wouldn't know whether to return the new enum or to return the `UNKNOWN` value to the caller.

Note that the previous version of this API didn't have an "Unknown" value. If an process exited which didn't have a kind assigned in v1, it just returned the v2 value for that process kind, and v1 callers just have to deal with the possibility that they can be given a kind "from the future". Can we just continue that pattern?